### PR TITLE
use binary searched data value for cursor info display

### LINKED ISSUE #1917: show exact data x and y values as cursor info in the histogram widget
**Describe the bug**
Currently the displayed cursor "x" value in the histogram widget is derived from the plot domain, not the data domain (ie display the xy values of the nearest data point). For practical usages, we should show the x value based on the data points.

https://user-images.githubusercontent.com/20819712/179893314-123e7f13-44ec-4493-8140-cef34322ce6b.mov

**To Reproduce**
1. load an image and launch the histogram widget
2. move cursor around on the histogram plot and observe the displayed cursor values
3. y value is derived from the y value of the data point. This is correct.
4. x value is not derived from the x value of the data point. Instead, it is derived from the plot domain. This is the issue.

**Expected behavior**
Cursor info in the histogram widget should show values of the nearest data point.

**Platform info (please complete the following information):**
 - OS [e.g. macOS Monterey]: macOS Monterey
 - Browser [e.g. chrome, safari, electron app]: chrome
 - Browser version [e.g. 22]: 103
 - Backend branch [e.g. dev, v3b2 release]: dev
 - Frontend branch [e.g. dev, v3b2 release]: dev

### PR DESCRIPTION
**Pull request by kswang1029**
**Description**
This tries to fix #1917 by adopting the binary searched data value (x and y) for cursor info display. The original displayed x value is derived from the coordinate system of the plot. 

**Checklist**
- [x] changelog updated / ~~no changelog update needed~~
- [x] e2e test passing / ~~added corresponding fix~~
- [x] ~~protobuf updated to the latest dev commit~~ / no protobuf update needed

### CODE REVIEW

File: src/components/Histogram/HistogramComponent.tsx [hunk 1]
```diff
@@ -194,30 +194,23 @@ export class HistogramComponent extends React.Component<WidgetProps> {
...
-                if (this.widgetStore.cursorX < 1e-5) {
-                    numberString = toExponential(this.widgetStore.cursorX, 5);
-                } else {
-                    numberString = toFixed(this.widgetStore.cursorX, 5);
-                }
                 const nearest = binarySearchByX(this.plotData.values, this.widgetStore.cursorX);
                 if (nearest?.point) {
-                    let valueLabel = `${nearest.point.y !== 0.5 ? nearest.point.y : 0}`;
-
+                    let xValueLabel = Math.abs(nearest.point.x) < 1e-5 ? `${toExponential(nearest.point.x, 5)}` : `${toFixed(nearest.point.x, 5)}`;
```
Comment (YuHsuan-Hwang) [inline src/components/Histogram/HistogramComponent.tsx]: One minor suggestion: we can remove `${...}` because `toExponential` and `toFixed` return string.

Comment (kswang1029) [reply src/components/Histogram/HistogramComponent.tsx]: corrected.

Comment (YuHsuan-Hwang) [reply src/components/Histogram/HistogramComponent.tsx]: Oh, I mean we can remove ' ${...} '

Comment (kswang1029) [reply src/components/Histogram/HistogramComponent.tsx]: corrected.

**Review by YuHsuan-Hwang — APPROVED** [review]

**Pull request merged**